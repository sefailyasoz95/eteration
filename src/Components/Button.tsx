import React from "react";
import {
	Text,
	TouchableOpacity,
	StyleSheet,
	StyleProp,
	ViewStyle,
	GestureResponderEvent,
	TextStyle,
	ColorPropType,
} from "react-native";

type Props = {
	text: string;
	buttonStyle?: StyleProp<ViewStyle>;
	textStyle?: StyleProp<TextStyle>;
	corner?: "cornered" | "curved" | "rounded" | number;
	type?: "outlined" | "filled";
	buttonColor?: string;
	textColor?: string;
	onPress: (event: GestureResponderEvent) => void;
};

const Button: React.FC<Props> = ({ text, type, buttonColor, textStyle, textColor, corner, buttonStyle, onPress }) => {
	const buttonTypes: Object = {
		borderRadius: corner === "cornered" ? 0 : corner === "curved" ? 5 : corner === "rounded" ? 10 : corner,
		backgroundColor: type === "filled" ? buttonColor : "rgba(0,0,0,0)",
		borderColor: buttonColor,
	};
	return (
		<TouchableOpacity style={[styles.baseContainerStyle, buttonStyle, buttonTypes]} onPress={onPress}>
			<Text style={[styles.baseTextStyle, textStyle, { color: textColor ? textColor : "#000" }]}>{text}</Text>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	baseContainerStyle: {
		padding: 5,
		borderWidth: 1,
	},
	baseTextStyle: {
		letterSpacing: 1,
		fontWeight: "500",
	},
});

export default Button;
