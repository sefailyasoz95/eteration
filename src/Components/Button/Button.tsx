import React from "react";
import {
	Text,
	TouchableOpacity,
	StyleSheet,
	StyleProp,
	ViewStyle,
	GestureResponderEvent,
	TextStyle,
	ColorPropType,
	ActivityIndicator,
} from "react-native";

type Props = {
	text: string;
	buttonStyle?: StyleProp<ViewStyle>;
	textStyle?: StyleProp<TextStyle>;
	corner?: "cornered" | "curved" | "rounded" | "circle" | number;
	type?: "none" | "outlined" | "filled";
	buttonColor?: string;
	textColor?: string;
	onPress: (event: GestureResponderEvent) => void;
	loading?: boolean;
	testID: string;
};

const Button: React.FC<Props> = ({
	text,
	type,
	buttonColor,
	textStyle,
	textColor,
	corner,
	buttonStyle,
	onPress,
	loading,
	testID,
}) => {
	const buttonTypes: Object = {
		borderRadius:
			corner === "cornered"
				? 0
				: corner === "curved"
				? 5
				: corner === "rounded"
				? 10
				: corner === "circle"
				? "50%"
				: corner,
		backgroundColor: type === "filled" ? buttonColor : "rgba(0,0,0,0)",
		borderColor: type === "none" ? "rgba(0,0,0,0)" : buttonColor,
		borderWidth: !type || type === "none" ? 0 : 1,
	};
	return (
		<TouchableOpacity style={[styles.baseContainerStyle, buttonStyle, buttonTypes]} onPress={onPress} testID={testID}>
			{loading ? (
				<ActivityIndicator color={textColor} />
			) : (
				<Text style={[styles.baseTextStyle, textStyle, { color: textColor ? textColor : "#000" }]}>{text}</Text>
			)}
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	baseContainerStyle: {
		padding: 5,
	},
	baseTextStyle: {
		letterSpacing: 1,
		fontWeight: "500",
	},
});

export default Button;
