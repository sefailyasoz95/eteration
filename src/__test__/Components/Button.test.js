import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Button from "../../Components/Button/Button";

Enzyme.configure({ adapter: new Adapter() });

describe("Button Component", () => {
	it("should render screen correctly", () => {
		const component = shallow(<Button />);
		expect(component.length).toBe(1);
		expect(toJson(component)).toMatchSnapshot();
	});
	it("should simulate the button press", () => {
		const onPressMock = jest.fn();
		const wrapper = shallow(<Button onPress={onPressMock} />);
		expect(wrapper.find({ testID: "customButton" }).simulate("click"));
		expect(onPressMock).toBeTruthy();
	});
});
