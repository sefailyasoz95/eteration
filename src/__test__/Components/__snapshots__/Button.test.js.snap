// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render screen correctly 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "padding": 5,
      },
      undefined,
      Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": 0,
      },
    ]
  }
  testID="customButton"
>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "500",
          "letterSpacing": 1,
        },
        undefined,
        Object {
          "color": "#000",
        },
      ]
    }
  />
</ForwardRef>
`;
